### PR TITLE
[FIX] base: link res_partner to their users

### PR DESCRIPTION
**Pull request by fdamhaut**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/base/models/res_users.py [hunk 1]
```diff
@@ -696,6 +696,7 @@ def create(self, vals_list):
         users = super(Users, self).create(vals_list)
         setting_vals = []
         for user in users:
+            user.partner_id.with_context(mail_auto_subscribe_no_notify=True).user_id = user
```
Comment (kmagusiak) [inline odoo/addons/base/models/res_users.py:699]: This is not correct. `user_id` is the sales representative for the partner, not the user linked to the partner. For this you have the `res.partner.user_ids` field which is the inverse of `res.users.partner_id`.

**Comment by tivisse** [comment]
@robodoo r+ rebase-ff

**Review by kmagusiak — CHANGES_REQUESTED** [review]
I think you have confused `user_id` and `user_ids` on the partner.

File: addons/website_sale_slides/tests/test_course_purchase_flow.py [hunk 2]
```diff
@@ -61,6 +61,7 @@ def test_course_purchase_flow(self):
         # Step 4: Same test as salesman
         salesman_sale_order = self.env['sale.order'].with_user(self.user_salesman).create({
             'partner_id': self.user_portal.partner_id.id,
+            'user_id': self.user_salesman.id,
```
Comment (tde-banana-odoo) [inline addons/website_sale_slides/tests/test_course_purchase_flow.py:64]: This should not be necessary, as there is a with_user used for the create. Why this change ?

File: odoo/addons/base/models/res_users.py (hunk 1 above)
Comment (tde-banana-odoo) [reply odoo/addons/base/models/res_users.py:699]: Yes, don't think we need user to have themselves as sales rep ...

**Review by tde-banana-odoo — COMMENTED** [review]
Do not r+ on Friday or with a sinusitis. That's a Chinese proverbe !

**Comment by tde-banana-odoo** [comment]
Thought it was merged, sorry.
